source/slang/slang-ast-support-types.h

```diff
@@ -167,6 +167,10 @@ namespace Slang
         kConversionCost_ScalarToMatrix = 10,
         kConversionCost_ScalarIntegerToFloatMatrix = kConversionCost_IntegerToFloatConversion + kConversionCost_ScalarToMatrix,
 
+        // Additional conversion cost to add when casting matrix to vector and vise versa
+        kConversionCost_MatrixToVector = 1,
```

Comment:
This cost seems too small.

Reply:
I will set it to 3:
1. more expensive than kConversionCost_OneVectorToScalar (2)
2. cheaper than a transpose on registers by a large amount (5)